### PR TITLE
Health Events module v1

### PR DESCRIPTION
**Pull request by esc1144**
*Issue #, if available:*

First release of Health Events module.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-collection/test/utils.py [hunk 1]
```diff
@@ -140,7 +140,9 @@ def deploy_stack(cloudformation, stack_name: str, file: Path, parameters: list[d
         Capabilities=['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM', 'CAPABILITY_AUTO_EXPAND'],
         Tags=[ {'Key': 'branch', 'Value': 'branch'},],
         NotificationARNs=[],
-        TemplateBody=file.open().read(),
+        # This allows us to deploy via a bucket and avoid the more limited max template length issue
```
Comment (iakov-aws) [inline data-collection/test/utils.py:143]: do we need to leave these comments ?

File: data-collection/deploy/module-health-events.yaml [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+
+          logger = logging.getLogger()
+          logger.setLevel(getattr(logging, os.environ.get('LOG_LEVEL', 'INFO').upper(), logging.INFO))
+
+          BUCKET_NAME = os.environ['BUCKET_NAME']
+          ROLENAME = os.environ['ROLENAME']
+          PREFIX = os.environ['PREFIX']
+          REGIONS = [r.strip() for r in os.environ.get("REGIONS", "").split(',') if r]
+          if len(REGIONS) > 0:
+              REGIONS.append('global')
```
Comment (iakov-aws) [inline data-collection/deploy/module-health-events.yaml:136]: can be simpler with ?

`REGIONS = [r.strip() for r in os.environ.get("REGIONS", "global").split(',')]`

File: data-collection/deploy/module-health-events.yaml [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+                      }
+                  }
+                  if len(REGIONS) > 0:
+                      args['filter']['regions'] = REGIONS
+
+                  ingestion_time = datetime.now(timezone.utc)
+                  try:
+                      with open(TMP_FILE, "w", encoding='utf-8') as f:
+                          f.write('eventArn,eventScopeCode\n')
+                          for _, h_event in enumerate(search(health_client.describe_events_for_organization, args, expression='events')):
```
Comment (iakov-aws) [inline data-collection/deploy/module-health-events.yaml:335]: do we need enumerate?

File: data-collection/test/utils.py [hunk 4]
```diff
@@ -286,6 +288,7 @@ def _extract_lambda_arns(state):
     last_log_time = {lambda_arn: int(time.time()) * 1000 for lambda_arn in lambda_arns}
     execution_results = {execution_arn: None for execution_arn in execution_arns}
     running = True
+    time.sleep(20) # should refactor to base on waiting on crawlers
     while running:
```
Comment (iakov-aws) [inline data-collection/test/utils.py:292]: do we still need this sleep?

**Review by iakov-aws — APPROVED** [review]

File: data-collection/deploy/module-health-events.yaml (hunk 2 above)
Comment (esc1144) [reply data-collection/deploy/module-health-events.yaml:136]: We need global in the list regardless. What I need to do is:
REGIONS = ['global']
REGIONS.extend([r.strip() for r in os.environ.get("REGIONS", "").split(',') if r])

File: data-collection/test/utils.py (hunk 4 above)
Comment (esc1144) [reply data-collection/test/utils.py:292]: As we add more modules, I am seeing a number of check fails only due to not enough time to catch up. There is throttling happening on the burst of crawler calls, which may be part of it. When there is time I think we should just rethink it a little to check the status.

File: data-collection/deploy/module-health-events.yaml (hunk 3 above)
Comment (esc1144) [reply data-collection/deploy/module-health-events.yaml:335]: I need to investigate for next release.

**Pull request merged**